scripts/pre-commit-markdown-quality.sh

```
@@ -0,0 +1,84 @@
+if [[ -z "$staged_md_files" ]]; then
+    echo -e "${GREEN}✅ No markdown files to check${NC}"
+    exit 0
+fi
+
+echo "Checking staged markdown files:"
+echo "$staged_md_files" | sed 's/^/  - /'
+echo
+
+# Create temporary directory for staged files
+temp_dir=$(mktemp -d)
+trap "rm -rf $temp_dir" EXIT
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)